### PR TITLE
wp popups beta

### PR DESCRIPTION
**Pull request by GunkaArtur**
Related issue: https://github.com/bagrinsergiu/blox-editor/issues/25612

### CODE REVIEW

File: public/editor-client/src/api/index.ts [hunk 1]
```diff
@@ -1206,3 +1214,60 @@ export const updateGlobalBlocks = async (
...
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    const parsedResult = mPipe(
+      pass(isPopupDataResult),
+      (res) => res.pop()?.pageData,
+      (r) => JSON.parse(r),
```
Comment (victor-pg) [inline public/editor-client/src/api/index.ts:1259]: use `Json.parse` from "@brizy/readers"

File: public/editor-client/src/defaultTemplates/utils.ts [hunk 2]
```diff
@@ -99,289 +107,94 @@ export const converterPopup = (
...
+  return (
+    Obj.isObject(obj) &&
+    Obj.hasKey("categories", obj) &&
+    Obj.hasKey("order", obj) &&
+    Obj.hasKey("pro", obj) &&
+    Obj.hasKey("thumbnail", obj) &&
+    Obj.hasKey("thumbnailHeight", obj) &&
+    Obj.hasKey("thumbnailWidth", obj) &&
+    Obj.hasKey("title", obj) &&
+    Obj.hasKey("id", obj)
```
Comment (victor-pg) [inline public/editor-client/src/defaultTemplates/utils.ts:151]: use `Obj.hasKeys`

File: public/editor-client/src/defaultTemplates/utils.ts [hunk 3]
```diff
@@ -99,289 +107,94 @@ export const converterPopup = (
...
+  );
+};
 
-  allElements = allElements.concat(firstPageResponse.collections);
+export const isPopupsResponse = (obj: unknown): obj is PopupsResponse => {
+  return (
+    Obj.isObject(obj) &&
+    Obj.hasKey("categories", obj) &&
+    Obj.hasKey("collections", obj) &&
```
Comment (victor-pg) [inline public/editor-client/src/defaultTemplates/utils.ts:159]: use `Obj.hasKeys`

File: public/editor-client/src/defaultTemplates/utils.ts [hunk 4]
```diff
@@ -99,289 +107,94 @@ export const converterPopup = (
...
+    Array.isArray(obj.categories) &&
+    obj.categories.every(
+      (item) =>
+        Obj.isObject(item) &&
+        Obj.hasKey("title", item) &&
+        Obj.hasKey("slug", item) &&
```
Comment (victor-pg) [inline public/editor-client/src/defaultTemplates/utils.ts:165]: use `Obj.hasKeys`

**Review by victor-pg — CHANGES_REQUESTED** [review]

File: public/editor-client/src/api/index.ts (hunk 1 above)
Comment (GunkaArtur) [reply public/editor-client/src/api/index.ts:1259]: we don't have parse 

image

Comment (victor-pg) [reply public/editor-client/src/api/index.ts:1430]: 1. yes, but we have `Json.read`, so use it
2. `JSON.parse` returns `any`, that's why you don't get error

File: public/editor-client/src/defaultTemplates/utils.ts [hunk 5]
```diff
@@ -153,6 +155,42 @@ export const convertLayoutPages = (
...
+      id,
+      title,
+      categories,
+      pro,
+      thumbnail,
+      thumbnailWidth,
+      thumbnailHeight,
+      blank
+    }) => ({
+      id: id,
```
Comment (victor-pg) [inline public/editor-client/src/defaultTemplates/utils.ts:175]: write short

File: public/editor-client/src/defaultTemplates/utils.ts [hunk 6]
```diff
@@ -153,6 +155,42 @@ export const convertLayoutPages = (
...
+      cat: categories.split(",").map((item) => item.trim().toLowerCase()),
+      title: title,
```
Comment (victor-pg) [inline public/editor-client/src/defaultTemplates/utils.ts:177]: write short

**Review by victor-pg — CHANGES_REQUESTED** [review]

**Review by maxval1 — CHANGES_REQUESTED** [review]

File: public/editor-client/src/defaultTemplates/index.ts [hunk 7]
```diff
@@ -64,9 +65,11 @@ const defaultKits = (
     },
     async getData(res, rej, { kitId, id }) {
       try {
-        const data = await getKitData(blocksDataUrl, kitId, id);
+        if (kitId) {
```
[the remark this answers is not in the document]
Comment (GunkaArtur) [reply public/editor-client/src/defaultTemplates/index.ts:68]: because Kits and Popups are almost the same 

Kits need kitId

Popups don't need it

**Pull request merged**